src/vivarium_csu_zenon/components/treatment/treatment_coverage.py

```diff
@@ -0,0 +1,148 @@
+                                                                   project_globals.IHD_MODEL_NAME],
+                                                 requires_values=['high_ldl_cholesterol.exposure'],
+                                                 requires_streams=[self.name])
+
+    def on_initialize_simulants(self, pop_data: 'SimulantData'):
+        """Initialize treatment status"""
+        pop_update = pd.DataFrame({
+            parameters.STATIN_HIGH: 'none',  # 'none', 'low', 'high'
+            parameters.STATIN_LOW: 'none',  # 'none', 'low', 'high'
+            parameters.EZETIMIBE: False,
+            parameters.FIBRATES: False,
+            parameters.LIFESTYLE: False,
+            parameters.FDC: False,
+        })
+
+        ldlc = self.ldlc(pop_data.index)
+        treatment_probability = self.get_treatment_probability(ldlc)
+        treated = self.randomness.get_draw(pop_data.index) < treatment_probability
+        mono_if_treated = (self.randomness.get_draw(pop_data.index, additional_key='therapy_type')
+                           < self.p_therapy_type[parameters.MONOTHERAPY])
+        treatment_type_if_mono = self.randomness.choice(pop_data.index,
+                                                        list(self.p_treatment_type.keys()),
+                                                        list(self.p_treatment_type.values()),
+                                                        additional_key='treatment_type')
+        ezetimibe_if_mono = treatment_type_if_mono == parameters.EZETIMIBE
+        fibrates_if_mono = treatment_type_if_mono == parameters.FIBRATES
+        low_statin_if_mono = treatment_type_if_mono == parameters.STATIN_LOW
+        high_statin_if_mono = treatment_type_if_mono == parameters.STATIN_HIGH
+        fdc_if_multi = (self.randomness.get_draw(pop_data.index, additional_key='fdc')
+                        < self.p_therapy_type[parameters.FDC])
+        p_low = (self.p_treatment_type[parameters.STATIN_LOW]
+                 / (self.p_treatment_type[parameters.STATIN_LOW] + self.p_treatment_type[parameters.STATIN_HIGH]))
+        # noinspection PyTypeChecker
+        low_potency_statin_if_not_fdc = (self.randomness.get_draw(pop_data.index, additional_key='low_potency_statin')
+                                         < p_low)
+        low_dose_if_statin = (self.randomness.get_draw(pop_data.index, additional_key='statin_dose')
+                              < parameters.PROB_LOW_DOSE)
+
+        # potency_statin_dose
+        high_statin = ((treated & mono_if_treated & high_statin_if_mono)
+                       | (treated & ~mono_if_treated & ~fdc_if_multi & ~low_potency_statin_if_not_fdc))
+        low_statin = ((treated & mono_if_treated & low_statin_if_mono)
+                      | (treated & ~mono_if_treated & ~fdc_if_multi & low_potency_statin_if_not_fdc))
+        # potency_statin_dose
```

Comment:
This may be wrong depending on how dosing of statins on FDC is supposed to be handled.